File: backend/IDE.Common/ModelsDTO/DTO/Workspace/FileDTO.cs
using IDE.Common.ModelsDTO.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace IDE.Common.ModelsDTO.DTO.Workspace
{
    public class FileDTO
    {
        public FileDTO()
        {
            NestedFiles = new List<FileDTO> ();
        }
        public string Id { get; set; }
        public TreeNodeType Type { get; set; }
        public string Name { get; set; }
        public string Details { get; set; }
        public ICollection<FileDTO> NestedFiles { get; set; }
    }
}

File: frontend/src/app/models/DTO/Workspace/fileDTO.ts
/* Auto Generated */

import { TreeNodeType } from "./../../Enums/treeNodeType"
import { FileDTO } from "./../File/fileDTO"

export interface FileDTO {
    id: string;
    type: TreeNodeType;
    name: string;
    details: string;
    nestedFiles: FileDTO[];
}
